src/Components/Blueprints/BuildImagesButton.tsx

```diff
@@ -1,20 +1,185 @@
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  Flex,
+  FlexItem,
+  Spinner,
+  MenuToggleAction,
+  ButtonProps,
+  Button,
+} from '@patternfly/react-core';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
+import { skipToken } from '@reduxjs/toolkit/query';
 
 import { selectSelectedBlueprintId } from '../../store/BlueprintSlice';
-import { useAppSelector } from '../../store/hooks';
-import { useComposeBlueprintMutation } from '../../store/imageBuilderApi';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import {
+  ImageTypes,
+  useComposeBlueprintMutation,
+  useGetBlueprintQuery,
+} from '../../store/imageBuilderApi';
 
-type BuildImagesButtonPropTypes = {
+export const BuildImagesButton = () => {
+  const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
+  const [deselectedTargets, setDeselectedTargets] = useState<ImageTypes[]>([]);
+  const [buildBlueprint, { isLoading: imageBuildLoading }] =
+    useComposeBlueprintMutation();
+  const dispatch = useAppDispatch();
+
+  const onBuildHandler = async () => {
+    if (selectedBlueprintId) {
+      try {
+        await buildBlueprint({
+          id: selectedBlueprintId,
+          body: {
+            image_types: blueprintImageType?.filter(
+              (target) => !deselectedTargets?.includes(target)
+            ),
+          },
+        });
+      } catch (imageBuildError) {
+        dispatch(
+          addNotification({
+            variant: 'warning',
+            title: 'No blueprint was build',
+          })
+        );
+      }
+    }
+  };
+  const [isOpen, setIsOpen] = useState(false);
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+  const { data: blueprintDetails } = useGetBlueprintQuery(
+    selectedBlueprintId ? { id: selectedBlueprintId } : skipToken
+  );
+  const blueprintImageType = blueprintDetails?.image_requests.map(
+    (image) => image.image_type
+  );
+
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent>,
+    itemId: number
+  ) => {
+    const imageType = blueprintImageType?.[itemId];
+
+    if (imageType && deselectedTargets?.includes(imageType)) {
+      setDeselectedTargets(
+        deselectedTargets.filter((target) => target !== imageType)
+      );
+    } else if (imageType) {
+      setDeselectedTargets([...deselectedTargets, imageType]);
+    }
+  };
+
+  const handleToggleEnvironment = (target: ImageTypes) => {
+    switch (target) {
+      case 'aws':
+        return 'Amazon Web Services (AWS)';
+      case 'gcp':
+        return 'Google Cloud Platform (GCP)';
+      case 'azure':
+        return 'Microsoft Azure';
+      case 'oci':
+        return 'Oracle Cloud Infrastructure';
+      case 'vsphere-ova':
+        return 'VMware vSphere';
+      case 'guest-image':
+        return 'Virtualization - Guest image (.qcow2)';
+      case 'image-installer':
+        return 'Bare metal - Installer';
+    }
+  };
```

Comment:
https://github.com/osbuild/image-builder-frontend/pull/2135 got merged today - you can use `targetOptions` const to replace this function so the naming is consistent throughout the code base: https://github.com/osbuild/image-builder-frontend/blob/ee216857926bf1a628e74733bd4bf7942b7ad4e3/src/constants.ts#L51-L67

Reply:
But we can split it to a follow up as well, whatever works better :) This is just a heads up